pkgs/development/libraries/boost/generic.nix

```diff
@@ -104,7 +104,10 @@ stdenv.mkDerivation {
   patchFlags = "";
 
   patches = patches
-    ++ optional stdenv.isDarwin ./darwin-no-system-python.patch;
+  ++ optional stdenv.isDarwin (
+    if version == "1.55.0"
+    then ./darwin-1.55-no-system-python.patch
+    else ./darwin-no-system-python.patch);
```

Comment:
Did you check any other boost versions? Should we move this out from "generic.nix" to "1.55.nix", "1.59.nix" and so on..?

Reply:
> Did you check any other boost versions?

Yup. They all build on my macOS 10.12 laptop:

    $ nix-build -A boost155 -A boost159 -A boost160 -A boost162 -A boost163 -A boost164 -A boost165 -A boost166 -A boost167 -A boost168 -A boost169

> Should we move this out from "generic.nix" to "1.55.nix", "1.59.nix" and so on..?

I'm fine doing that, but I think that'd end up dirtier than what I implemented here.